error.go

```diff
@@ -44,8 +44,13 @@ type ErrorDetail struct {
 	wireJSON string // preserve human-readable JSON
 }
 
-// NewErrorDetail constructs a new error detail.
+// NewErrorDetail constructs a new error detail. This returns an error
+// if msg cannot be marshalled into an Any message.
```

Comment:
A surprising number of people aren't familiar with the WKTs. Can we turn `Any` into a newfangled GoDoc link to the docs for `*anypb.Any`? I think `[google.golang.org/protobuf/types/known/anypb.Any]` works.

Also, can we document that this doesn't re-wrap if you pass an Any?

Reply:
Done. I've used simple type links, so I changed `Any` to `[anypb.Any]`.

An alternative would be to write it as `[Any]` and then add a footnote:
```
[Any]: https://pkg.go.dev/google.golang.org/protobuf/types/known/anypb#Any
```
Would you prefer that alternative? ([reference](https://tip.golang.org/doc/comment#links))